inttest/footloose-alpine/Dockerfile

```diff
@@ -27,27 +52,32 @@ RUN sed -i -e 's/^\(tty[0-9]\)/# \1/' /etc/inittab
 RUN sed -i -e 's/^root:!:/root::/' /etc/shadow
 
 # Put kubectl into place to ease up debugging
-RUN curl -Lo /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v$KUBE_VERSION/bin/linux/amd64/kubectl \
+RUN curl -Lo /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v$KUBE_VERSION/bin/linux/$ARCH/kubectl \
   && chmod +x /usr/local/bin/kubectl
 ENV KUBECONFIG=/var/lib/k0s/pki/admin.conf
 
 # Install troublbeshoot support bundle
-RUN curl -Lo - https://github.com/replicatedhq/troubleshoot/releases/download/v0.55.0/support-bundle_linux_amd64.tar.gz \
-  | tar xzO support-bundle >/usr/local/bin/kubectl-supportbundle \
-  && chmod +x /usr/local/bin/kubectl-supportbundle
+COPY --from=supportbundle out/* /usr/local/bin/
 
 # Put helm into place to ease up debugging and for helm integration tests
-RUN curl -L https://get.helm.sh/helm-v$HELM_VERSION-linux-amd64.tar.gz \
-  | tar xz -C /usr/local/bin --strip-components=1 && chmod +x /usr/local/bin/helm
+RUN curl -L https://get.helm.sh/helm-v$HELM_VERSION-linux-$ARCH.tar.gz \
+  | tar xz linux-$ARCH/helm -C /usr/local/bin --strip-components=1 \
+  && chmod +x /usr/local/bin/helm
 
 # Install etcd for smoke tests with external etcd
-RUN curl -L https://github.com/etcd-io/etcd/releases/download/v$ETCD_VERSION/etcd-v$ETCD_VERSION-linux-$ETCD_ARCH.tar.gz \
-  | tar xz -C /opt --strip-components=1
+# No arm binaries available (check-externaletcd won't work on ARMv7)
+RUN [ "$ARCH" == arm ] || { \
+  curl -L https://github.com/etcd-io/etcd/releases/download/v$ETCD_VERSION/etcd-v$ETCD_VERSION-linux-$ARCH.tar.gz \
+    | tar xz -C /opt --strip-components=1; \
+}
 
 # Install cri-dockerd shim for custom CRI testing
-RUN curl -sSfLo /tmp/cri-dockerd.tgz https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.1/cri-dockerd-0.3.1.amd64.tgz \
-  && tar xf /tmp/cri-dockerd.tgz --directory /tmp/ \
-  && mv /tmp/cri-dockerd/cri-dockerd /usr/local/bin/cri-dockerd \
-  && rm -rf /tmp/cri-dockerd \
-  && chmod 755 /usr/local/bin/cri-dockerd
+# No arm binaries available (check-byocri won't work on ARMv7)
+RUN [ "$ARCH" == arm ] || { \
+  curl -sSfLo /tmp/cri-dockerd.tgz https://github.com/Mirantis/cri-dockerd/releases/download/v0.3.1/cri-dockerd-0.3.1.$ARCH.tgz \
```

Comment:
I know this is not introduced here but since we have other versions passed as arguments, I think this should be passed as an argument too.

Reply:
Do you mean the version number? It has been hard coded since its inception, I think.

Question is why we want to do it. It's nowhere used outside this file. The version numbers that are actually picked up from variables are usually shared with other parts of the build. For the sake of grouping version numbers together, I'd rather put those a bit higher up in the Dockerfile and put a default on them. Same for the troubleshoot version. We can still move them to another location when they actually need to be shared with something else, I'd say.

Reply:
Fair enough.